js_modules/dagit/packages/core/src/app/Permissions.tsx

```diff
@@ -74,17 +74,32 @@ const extractPermissions = (permissions: PermissionFragment[]) => {
 
 export type PermissionsMap = ReturnType<typeof extractPermissions>;
 
-export const PermissionsContext = React.createContext<PermissionFragment[]>([]);
+export const PermissionsContext = React.createContext<{
+  data: PermissionFragment[];
+  loading: boolean;
+}>({data: [], loading: true});
 
 export const PermissionsProvider: React.FC = (props) => {
-  const {data} = useQuery<PermissionsQuery>(PERMISSIONS_QUERY);
-  const value = React.useMemo(() => data?.permissions || [], [data]);
+  const {data, loading} = useQuery<PermissionsQuery>(PERMISSIONS_QUERY);
+  const value = React.useMemo(
+    () => ({
+      data: data?.permissions || [],
+      loading,
+    }),
+    [data, loading],
+  );
   return <PermissionsContext.Provider value={value}>{props.children}</PermissionsContext.Provider>;
 };
 
 export const usePermissions = () => {
-  const rawPermissions = React.useContext(PermissionsContext);
-  return React.useMemo(() => extractPermissions(rawPermissions), [rawPermissions]);
+  const {data, loading} = React.useContext(PermissionsContext);
+  return React.useMemo(
+    () => ({
+      ...extractPermissions(data),
+      loading,
```

Comment:
Seems like `loading` would work better alongside the permissions object, rather than part of it.

```
{
  permissions: extractPermissions(data),
  loading,
}
```

I'm guessing that this would be annoying right now because all the callsites would need to be updated to extract from the `permissions` value? If so, it would be nice to do the refactor in a followup.

Reply:
Yeah it's annoying but its also not a problem, everything is well typed